lib/chef/knife/winrm_base.rb

```diff
@@ -114,7 +114,9 @@ def self.included(includer)
 
       def locate_config_value(key)
         key = key.to_sym
-        config[key] || Chef::Config[:knife][key]
+        value = config[key] || Chef::Config[:knife][key] || default_config[key]
```

Comment:
This entire method makes me nervous, or makes my scars from previous battles with argument precedence tingle.

Can't we just trust config[key] because it gets set similarly in Chef::Knife#merge_configs? https://github.com/chef/chef/blob/master/lib/chef/knife.rb#L341